Reorganizing Android Browser versions

afmenez opened the pull request:
- Adding metadata for all versions that had a browser (no results yet);
- Obsoleting ancient versions;
- Removing newer Android versions that are just aliases for Google Chrome (>4);

Commit:
Removing Android versions that are just aliases for Google Chrome

afmenez commented:
This patch addresses #1089 .

mgol commented:
What about Android Browser 4.2 & 4.3?

afmenez commented:
It's complicated.  As of Android 4.1, Chrome became the default Android browser.  But it is still possible to build the Android Browser unrtil 4.4, and some distros like Cyanogen do.  But it's the same version that was in 4.1.
This link has a decent explanation: https://decadecity.net/blog/2013/11/21/android-browser-versions

mgol commented:
I don't know if there are any differences between Android Browser in Android 4.2 & 4.3 but Can I Use reports some differences between 4.1 & 4.2/4.3: https://caniuse.com/#compare=android+4.1,android+4.2-4.3

So at least Android 4.2 should still be included.

mgol commented:
Another issue is that [babel-preset-env](https://www.npmjs.com/package/babel-preset-env) uses `compat-table` data so if you don't include Android 4.2 or 4.3, an entry like `Android >=4.3` will generate a bundle that may not compile in Android 4.3.

I think we should just include every browser version and - in particular - never remove existing ones from the data. Android 4.2 & 4.3 were present so it's best to not remove them.

Commit:
Adding missing Android Browser versions

afmenez commented:
OK, I have added all the missing versions up to Android 4.4.  I don't see the point of faking versions 5.0 and beyond, since it is just the Chrome browser.

mgol commented:
Yeah, I don't care that much about versions 4.4 & newer as they're based on Chrome. My remark was relevant for versions up to 4.3.

mgol commented on environments.json:
```diff
@@ -1595,50 +1595,117 @@
   },
   "android4_4": {
-    "full": "Android Browser",
+    "full": "Android Browser 4.4 (KitKat)",
```
I don't think this is correct. Android Browser ended with Android 4.3.

Also, it lacks  the `"equals": "chrome30"` declaration which was previously present. Doesn't this break correctness of the results?

As for me, Android 4.4 & newer was never correct as there's no Android Browser in Android 4.4 & newer so I'd be fine with removing them (If maintainers agree).

afmenez replied:
The info I have is that Chrome became the default browser for Android on JellyBean (4.1), and that's why I had removed 4.1-4.4 initially.
But my info also says that the sources were still available until KitKat (4.4), and some distros shipped with it.  I checked the link that you provided, and it mentions activity on 4.4 too (https://caniuse.com/#compare=android+4,android+4.1,android+4.2-4.3,android+4.4&compare_cats=JS).  And that's why I added them back.

mgol replied:
What Can I Use names as Android 4.4 browser is actually Chrome. You can see they have a separate entry for 4.4.3-4.4.4 as that's when the Chromium-based (!) WebView was updated to a newer Chromium version.

The link you provided at the beginning also says that the WebView in Android 4.4 is Chromium-based and that there is no "Browser" browser there.

afmenez replied:
Done.

afmenez commented:
The info I have is that Chrome became the default browser for Android on
JellyBean (4.1), and that's why I had removed 4.1-4.4 initially.
But my info also says that the sources were still available until KitKat
(4.4), and some distros shipped with it.  I checked the link that you
provided, and it mentions activity on 4.4 too (
https://caniuse.com/#compare=android+4,android+4.1,android+4.2-4.3,android+4.4&compare_cats=JS).
And that's why I added them back.

On Tue, Jun 6, 2017 at 7:02 PM, Michał Gołębiowski <notifications@github.com
> wrote:

> *@mgol* requested changes on this pull request.
> ------------------------------
>
> In environments.json
> <https://github.com/kangax/compat-table/pull/1104#discussion_r120493030>:
>
> >    },
>    "android4_4": {
> -    "full": "Android Browser",
> +    "full": "Android Browser 4.4 (KitKat)",
>
> I don't think this is correct. Android Browser ended with Android 4.3.
>
> Also, it lacks the "equals": "chrome30" declaration which was previously
> present. Doesn't this break correctness of the results?
>
> As for me, Android 4.4 & newer was never correct as there's no Android
> Browser in Android 4.4 & newer so I'd be fine with removing them (If
> maintainers agree).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kangax/compat-table/pull/1104#pullrequestreview-42460507>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABRIeEuXYOtOlpMzFRxEly1jOHI3j8qOks5sBcyJgaJpZM4NcgMr>
> .
>

Commit:
Removing Android 4.4 KitKat

chicoxyzzy commented:
@mgol do you approve this PR now?

mgol reviewed (approved):
Looks good! Thanks for listening me out. :)
